File: two-pointers/3sum.py
class Solution(object):
    def threeSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        nums.sort()
        n = len(nums)
        res = []

        for i in range(n): 
            if i > 0 and nums[i] == nums[i - 1]: 
                continue 
            target = -nums[i]
            l, r = i + 1, n-1

            while l < r : 
                s = nums[l] + nums[r]
                if s == target: 
                    res.append([nums[i], nums[l], nums[r]])

                    l_val, r_val = nums[l], nums[r]

                    while l < r and nums[l] == l_val: 
                        l += 1 
                    while l < r and nums[r] == r_val: 
                        r -= 1 

                elif s < target: 
                    l += 1 
                else: 
                    r -= 1 

        return res 

File: two-pointers/3sum.java
class Solution {
    public ArrayList<int[]> threeSum(int[] nums) {
    ArrayList<int[]> outputs = new ArrayList<int[]>();
    for(int i = 0; i < nums.length-2; i+=1) {
        for (int j = i +1; j <nums.length - 1; j+=1) {
          for (int k = i +1; j <nums.length -2; k+=1) {
            Boolean isvalid = false;
            if (nums[i]!=j & nums[i]!= k & nums[j]!=k) {
              if (nums[i] + nums[j] + nums[k] == 0) {
                isvalid= true;
                if (isvalid) {
                int valiseq[];
                int[] validseq = null;
				        validseq[0] = i;
                validseq[1] = j;
                validseq[2] = k;
                outputs.add(validseq);
                }
                
              }
            }
          }
        }
    }
    return outputs;
  }
}